### ISSUE TITLE
A class name was used instead of an instance name...

### ISSUE THREAD

**Issue by m-montagne** [dotnet/docs#26433]
While it is possible to understand why this error is raised in the case of my coding attempt, the raising of this error is uninstructive insofar as achieving a goal which I find solved in simpler terms, elsewhere on the internet: Specifically, I am attempting to build a List of _Types_, to be passed as arguments to the Add method of a page control. I find seemingly sound answers where Types are passed as arguments declared for instance as (Type type). Though I haven't tested the particular implementations because my need is more complex, the answers have been accepted. I understand that there are other ways I can do this, but the way I was hoping to provide the option of adding pages of different types was via a pair of lists -- one of string names, and the other of the related types.

So, the following was to be my list of _Types_:

		/* public static List<Type> PageTypesList()
			{
			List<Type> pageTypesList = new List<Type>();

			pageTypesList.Add(Page__BaseClass); // LIST LIKES AN INSTANCE, BUT NOT A TYPE. MAYBE LIST IS JUST WIRED THAT WAY, AND A COLLECTION MIGHT HOST A TYPE?
			pageTypesList.Add(Page_CRTable);
			pageTypesList.Add(Page_DBGridWithTools);
			pageTypesList.Add(Page_mfrCAM);
			pageTypesList.Add(Page_ResultSet);
			pageTypesList.Add(Page_ResultSetAndGrid);
			pageTypesList.Add(Page_SidePanelAndStatus);
			pageTypesList.Add(Page_SidePanelwSplitterAndStatus);
			pageTypesList.Add(Page_Status);
			pageTypesList.Add(Page_wTitle);

			return pageTypesList;
			} NO INDICATION FROM IDE WHY THESE TYPES OR SYNTAX ARE NOT ACCEPTABLE; "class" IS REJECTED AS WELL. */

Evidently then, List.Add() is _inherently_ looking for an _instance_; and though the IDE allows me to declare a list of List<Type>, List is incapable of adding a type (only) to the list array.

I can understand raising the error in regard to the usual purposes of Lists as references to instances. But I believe I have a marginally valid point (lol) in wanting a more instructive error message, because it is a possibly legitimate object to want to create some sort of array of _Types_.

Would it be too much to ask then, to desire further explanation that if the intention was to develop an array of _Types_... to explain that this must be done with a different vehicle... or cannot be done at all (because blah, blah, blah)?

I understand that this a fine point of contest, but please understand that given the possibly legitimate intention, a more instructive "error" message would do much to expedite resolution of a possibly legitimate (though admittedly not so common) object.

Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: eec72ac2-345b-8257-9765-426726d3ac40
* Version Independent ID: 8f8bdfec-3503-2b36-8d67-aa4dbf253c3c
* Content: [Compiler Error CS0119](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0119?f1url=%3FappId%3Droslyn%26k%3Dk(CS0119))
* Content Source: [docs/csharp/misc/cs0119.md](https://github.com/dotnet/docs/blob/main/docs/csharp/misc/cs0119.md)
* Product: **dotnet-csharp**
* Technology: **csharp-diagnostics**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Related to #24002
